### PR TITLE
Feat/hakyll projects page migration

### PR DESCRIPTION
**Pull request by Maeevick**
__Topic :__ Migration for the [__projects__ page](https://haskell.foundation/projects/) from hardcoded html to metadata in markdown.

__Status :__ _MVP / In Review_

__Steps :__ 

- _Create Projects Metadata_
- _Project list (html template)_
- _Project (html reusable partial template)_
- _Hakyll metadata parsing (md -> ctx)_
- _Partitioning according to status_
- _Hakyll generation (ctx -> html)_

__Possible Improvements :__ 

- __Extract leaders from metadata list__ _(temporary hardcoded solution)_

### CODE REVIEW

**Comment by Maeevick** [comment]
@myShoggoth I'm not satisfied about my leaders solution but you may be interested to take a look :smiley:

File: site.hs [hunk 1]
```diff
@@ -25,11 +25,10 @@ main = hakyll $ do
...
-                >>= applyAsTemplate defaultContext
-                >>= loadAndApplyTemplate "templates/boilerplate.html" defaultContext
-                >>= relativizeUrls
+        compile $ getResourceBody
+            >>= applyAsTemplate defaultContext
+            >>= loadAndApplyTemplate "templates/boilerplate.html" defaultContext
+            >>= relativizeUrls
 
     match "affiliates/*.markdown" $ do
         route $ setExtension "html"
```
Comment (Maeevick) [inline site.hs:34]: @myShoggoth I think we can remove this routing to affiliates pages, I don't see any access currently on prod

File: projects/ghc-performance-dashboards.markdown [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+status: ideation
+title: GHC Performance Dashboards
+summary: The Haskell Foundation is working with the GHC team to help provide dashboard monitors for Haskell critical libraries (e.g. Aeson, Text, Bytestring).
+link: https://gitlab.haskell.org/ghc/ghc
+link-text: https://gitlab.haskell.org/ghc/ghc
+subtitle: Project Leaders
+leader0name: Ben Gamari
+leader0img: /assets/images/project-leads/bg.png
```
Comment (myShoggoth) [inline projects/ghc-performance-dashboards.markdown:9]: Something I was not sure about was putting full image paths vs. knowing where they have to be for the context and just having the name.  So:

`leader0img: /assets/images/projectleads/bg.png`

vs. 

`leader0img: bg.png`

where the usage prepends the canonical path depending on the usage context. I went with the former but I think I'm leaning towards the latter. Not something we need to fix for this PR, just thought about it again while review.

File: projects/ghc-performance-dashboards.markdown [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+status: ideation
+title: GHC Performance Dashboards
+summary: The Haskell Foundation is working with the GHC team to help provide dashboard monitors for Haskell critical libraries (e.g. Aeson, Text, Bytestring).
```
Comment (myShoggoth) [inline projects/ghc-performance-dashboards.markdown:4]: I think if we're not going to have stand alone pages then the 'summary' can be the body text, which would also allow us to style it if appropriate.

**Review by myShoggoth — CHANGES_REQUESTED** [review]
It looks great, a few small things.

File: projects/ghc-performance-dashboards.markdown (hunk 2 above)
Comment (Maeevick) [reply projects/ghc-performance-dashboards.markdown:9]: problem with this is that some comes from "project-leads" and others come from "board-members" so it won't be only the name

```
leader0img: bg.png
leader1img: thc.png
```
vs
```
leader0img: project-leads/bg.png
leader1img: board-members/thc.png
```
We can image to separate the two properties : 

```
leader0team: project-leads
leader0img: bg.png
leader1team: board-members
leader1img: thc.png
```
and to build it on use in project template ```project.html```
```
<img src="$leader0team$/$leader0img$" alt="$leader0name$">
```
but it's maybe less straightforward than full path or team + name partial path

File: projects/ghc-performance-dashboards.markdown (hunk 3 above)
Comment (Maeevick) [reply projects/ghc-performance-dashboards.markdown:4]: I was thinking from the opposite :smile: 

If we want dedicated pages about each projects : the current previews are generated from metadata and each detailed page will be from body.

I'm taking a look, I spotted a problem with contexts and body in this part

File: projects/ghc-performance-dashboards.markdown (hunk 2 above)
Comment (Maeevick) [reply projects/ghc-performance-dashboards.markdown]: I let you look at the current proposal (compromise :smile:) @myShoggoth

File: projects/ghc-performance-dashboards.markdown (hunk 3 above)
Comment (Maeevick) [reply projects/ghc-performance-dashboards.markdown]: Done (if I had a good understanding of your wish :wink:)

**Review by myShoggoth — APPROVED** [review]
LGTM, thank you!

**Pull request merged**